pkg/appconsts/appconsts.go

```diff
@@ -49,21 +49,25 @@ const (
 	// in a continuation sparse share of a sequence.
 	ContinuationSparseShareContentSize = ShareSize - NamespaceSize - ShareInfoBytes
 
-	// DefaultMaxSquareSize is the maximum number of
-	// rows/columns of the original data shares in square layout.
-	// Corresponds to AVAILABLE_DATA_ORIGINAL_SQUARE_MAX in the spec.
-	// 128*128*256 = 4 Megabytes
-	// TODO(ismail): settle on a proper max square
-	// if the square size is larger than this, the block producer will panic
+	// DefaultMaxSquareSize is the maximum original square width. A square size
+	// larger than this will cause the block producer to panic.
+	//
+	// Note: 128 shares in a row * 128 shares in a column * 512 bytes in a share
+	// = 8 MiB
 	DefaultMaxSquareSize = 128
-	// MaxShareCount is the maximum number of shares allowed in the original data square.
-	// if there are more shares than this, the block producer will panic.
+
+	// MaxShareCount is the maximum number of shares allowed in the original
+	// data square. If there are more shares than this, the block producer will
+	// panic.
 	MaxShareCount = DefaultMaxSquareSize * DefaultMaxSquareSize
 
-	// DefaultMinSquareSize depicts the smallest original square width. A square size smaller than this will
-	// cause block producer to panic
+	// DefaultMinSquareSize is the smallest original square width. A square size
+	// smaller than this will cause the block producer to panic.
 	DefaultMinSquareSize = 1
-	// MinshareCount is the minimum shares required in an original data square.
+
+	// MinshareCount is the minimum number of shares allowed in the original
+	// data square. If there are fewer shares than this, the block producer will
+	// panic.
 	MinShareCount = DefaultMinSquareSize * DefaultMinSquareSize
```

Comment:
Why do we have a minimum?

Reply:
to generate the minimum data availability header, but tbh, we can probably just remove and calculate the value there

Reply:
Will create a follow-up issue.

1. `MaxShareCount` appears unused so it can likely be removed
2. `MinShareCount` can be removed from appconsts if we inline it https://github.com/rootulp/celestia-app/blob/51e826de5358542a797890ad9b26bbfe0e6b7c8f/pkg/da/data_availability_header.go#L178

Reply:
Created https://github.com/celestiaorg/celestia-app/issues/1183

Reply:
What about empty blocks?